src/r0/membership/invite_user.rs

```diff
@@ -18,8 +27,17 @@ ruma_api! {
         #[ruma_api(path)]
         pub room_id: RoomId,
         /// The user to invite.
-        pub user_id: UserId,
+        pub recipient: InvitationRecipient,
     }
 
     response {}
 }
+
+/// Distinguishes between invititations by Matrix or third party identifiers.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub enum InvitationRecipient {
+    /// Used to invite user by their Matrix identifer.
+    UserId(UserId),
+    /// Used to invite user by their third party identifer.
```

Comment:
```suggestion
    /// Used to invite user by a third party identifer.
```

A user can have multiple third party identifiers :)